File: problems/71-simplify-path/solution.go
package main

import "strings"

func simplifyPath(path string) string {
	var sections []string = strings.Split(path, "/")
	var res []string = []string{}

	for _, iV := range sections {
		if len(iV) < 1 || iV == "." {
			continue
		}

		if iV == ".." {
			if len(res) >= 1 {
				res = res[:len(res)-1]
			}
		} else {
			res = append(res, iV)
		}
	}

	return "/" + strings.Join(res, "/")
}

File: problems/71-simplify-path/solution.ts
function simplifyPath(path: string): string {
  if (path.length <= 1) return path;

  const pathSection = path.split("/");
  const res = [];

  for (let i = 0; i < pathSection.length; i++) {
    if (pathSection[i].length < 1) continue;

    switch (pathSection[i]) {
      case ".":
        continue;
      case "..":
        res.pop();
        continue;
      default:
        res.push(pathSection[i]);
    }
  }

  return `/${res.join("/")}`;
}

export { simplifyPath };
